sdk/core/core-http/test/defaultHttpClientTests.node.ts

```diff
@@ -40,6 +40,7 @@ describe("defaultHttpClient (node)", function() {
     });
 
     const client = new DefaultHttpClient();
+    (client as any)._fetch = httpMock.getFetch();
```

Comment:
I originally added a `DefaultHttpClient` constructor to allow passing in the mocked fetch, but felt changing public api just for testing seems too much.

Reply:
Honestly I think I'd like the constructor parameter better, since at the moment it's very hard to find all the places that set the secret property, since TS can't alias through `any` and the language service will claim it's never set.

Reply:
Changed to allow passing in fetch as a constructor parameter.

Reply:
Changed again.  Now mocking the `fetch()` instance method.